### ISSUE TITLE
False negative on `CanBeNonNullable`

### ISSUE THREAD

**Issue by atulgpt** [detekt/detekt#8498]
## Expected Behavior
```kt
@Test
fun `doesn't report for local property`() {
    val code = """
        fun baz() {
            var g: Int? = 1
        }
    """.trimIndent()
    assertThat(subject.lintWithContext(env, code)).hasSize(1)
}
```
Above TC should pass

## Observed Behavior
Above TC is failing

## Context
PR comment: https://github.com/detekt/detekt/pull/8496#discussion_r2285927923
